core/tracepoint/path_spec.rb

```diff
@@ -0,0 +1,20 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+
+ruby_version_is '2.0' do
+  describe 'TracePoint#path' do
+    it 'returns the path of the file being run' do
+      path = nil
+      TracePoint.new(:line) { |tp| path = tp.path }.enable do
+        path.should == "#{__FILE__}"
+      end
+    end
+
+    it 'equals (eval) inside an eval' do
+      path = nil
+      TracePoint.new(:end) { |tp| path = tp.path }.enable do
+        eval("class A; end")
```

Comment:
Ideally we wouldn't create a new global class...

Reply:
How about defining a method in eval? --
`eval("def test; end; test")`

Reply:
`eval("2 + 2")`?

Reply:
I tried this but I get file path instead of "(eval)" for line event -
```
Expected "/home/atul/spec/core/tracepoint/path_spec.rb"
to equal "(eval)"
```

Reply:
Best to use a different event that would happen only in the `eval`ed code.

Reply:
okay, I have updated the specs

Reply:
Please don't create a global method, though...